robottelo/ui/contentviews.py

```diff
@@ -450,6 +462,45 @@ def add_remove_errata_to_filter(self, cv_name, filter_name,
         else:
             self.click(locators['contentviews.remove_errata'])
 
+    def edit_erratum_date_range_filter(
+            self, cv_name, filter_name, errata_types=None, date_type=None,
+            start_date=None, end_date=None, open_filter=True):
+        """Edit Erratum Date Range Filter"""
+        allowed_errata_types = FILTER_ERRATA_TYPE.values()
+        allowed_date_types = FILTER_ERRATA_DATE.values()
+        erratum_type_locator = locators['contentviews.erratum_type_checkbox']
+        date_type_locator = locators['contentviews.erratum_date_type']
+        if open_filter:
+            self.go_to_filter_page(cv_name, filter_name)
+        if errata_types is not None:
+            if errata_types:
+                # because of the behaviour of the UI to check and disable
+                # the last element if the others are unchecked,
+                # will check all the elements first,
+                # after then uncheck the not selected ones
+                for errata_type in allowed_errata_types:
+                    self.assign_value(erratum_type_locator % errata_type, True)
+                for errata_type in allowed_errata_types:
+                    if errata_type not in errata_types:
+                        self.assign_value(
+                            erratum_type_locator % errata_type, False)
```

Comment:
Ok, so after discussion it appears that the last checked off checkbox can't be unchecked - it becomes disabled till you check off something else. So if we had `bugfix` errata and wan't to update it to `enhancement` - we shall check off enhancement first and only then uncheck bugfix. That explains 2 loops here.
With that said, i can see 2 solutions here:
1) 2 loops, one checking off elements, second one is unchecking. This is basically what's applied here, my only suggestion is in the first loop not to check off everything, but only those which need to be checked.
2) Form a list of checkboxes in the way that the ones which need to be checked will appear first. This way we can use only 1 loop:
```python
for errata in errata_types + list(set(allowed_errata_types).difference(errata_types)):
    self.assign_value(
        locators['contentviews.erratum_type_checkbox'] % checkbox,
        errata in errata_types
    )
```
The only problem with this approach - it looks a bit overcomplicated. Or it's just me? Any other reviewer's thoughts?